### ISSUE TITLE
Misreporting method naming should be snake_case for utf-8 method names

### ISSUE THREAD

**Issue by abinoam** [rubocop-hq/rubocop#6183]
## Expected behavior

I expect the following code snippet to not trigger method name snake_case cop.

```ruby
def ação(param)
end
```

## Actual behavior

Rubocop is (probably) misinterpreting utf-8 method names as non snake_case method names.

## Steps to reproduce the problem

Rubocop the above code snippet.

## RuboCop version

```
$ rubocop -V
0.58.2 (using Parser 2.5.1.2, running on ruby 2.5.1 x86_64-darwin17)
```

**Comment by martincabrera** [comment]
Working on it!

**Comment by martincabrera** [comment]
Proposed PR [here](https://github.com/rubocop-hq/rubocop/pull/6282)

**Comment by abinoam** [comment]
Great to see this PR @martincabrera !!!
I hope it gets merged as soon as possible.
